### ISSUE TITLE
Various profile issues caused by missing E-Mail address

### ISSUE THREAD

**Issue by subesokun** [RocketChat/Rocket.Chat#9390]
### Description:

Our LDAP server doesn't contain any E-Mail information and we're using SSO via LDAP for login. Hence when a user tries to login for the first time a new account gets created within RC which doesn't have any E-Mail information. As we're no admins of the LDAP server we can't just add the missing E-Mail information there.

Now the missing E-Mail information seems to cause various issues within RC. The profile page can't be loaded and also the My Profile > Settings and Security settings can't be changed. For debugging purposes we've manually added the E-Mail information to one of our users and then the profile page was accessible again.

Furthermore our usernames have the form `ntuser@company.com` (which is not our actual E-Mail address...) but the profile page is complaining about the `@` symbol in the username and greys out the save button so no changes can be made to the profile. Is there a way to strip away `@company.com` suffix from the RC username when doing SSO via LDAP?

### Server Setup Information:

* Version of Rocket.Chat Server: 0.60.1

### Steps to Reproduce:

1. Go to My Account
2. Select Profile

### Expected behavior:

Profile page gets shown and is editable

### Actual behavior:

Error gets thrown in the console and profile page doesn't get shown.

**Comment by subesokun** [comment]
Sorry some wrong information from my side. We're using SSO via Oauth against an Active Directory.

**Comment by timkinnane** [comment]
Yep, this is real. Happens for users created by any auth providers that don't have emails, even Facebook users can come through with just a phone number now. It also happens to the initial created admin, if they didn't set an email in environment configs.

There's two possible solutions.

1st. Force created users to get an empty email.

In [lib/accounts.js](https://github.com/RocketChat/Rocket.Chat/blob/develop/server/lib/accounts.js#L100-L160) `insertUserDoc`

Something like:
```
if (!Array.isArray(user.emails)) user.emails = [{ address: '', verified: true }]
   Meteor.users.update(userId, { $set: { emails } })
}
```

Then at least client code that expects it to be an array won't fail. When profile is opened, user will be forced to enter a new address, that will trigger proper verification. Not ideal, but less critical bug.
Also set to verified to prevent sending verification to non existent address.

2nd. Fix all the client code that assumes `user.emails` is an array.

Such as the example from this report in [accountProfile.js](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-ui-account/client/accountProfile.js#L103).

This would be cleaner. I don't think emails should be required. But could be an endless bug hunt?

**Comment by timkinnane** [comment]
Edit to above solutions. The empty email doesn't work, because I found there's also a **unique** index on the email field by default when using the Meteor Accounts package. So it requires actually dropping the email indexes on the model after accounts package has set them.

**Comment by Shailesh351** [comment]
After recent refactor of Profile Page, because of this missing email, it is not loading profile page at all.

<img width="1791" alt="Screenshot 2021-06-22 at 9 19 22 AM" src="https://user-images.githubusercontent.com/18264684/122860773-f44ab500-d33b-11eb-8c93-84a32619173e.png">

I'm creating PR for adding validation there 😄
